File: Array_I/Q1. Concatenation of Array/Q1_ConcatenationofArray.java
// Solution class
class Solution {

    // Method to concatenate array with itself
    public int[] getConcatenation(int[] nums) {

        // Store length of input array
        int len = nums.length;

        // Create new array of size 2 * len
        int[] ans = new int[2 * len];

        // Loop through original array
        for (int i = 0; i < len; i++) {

            // Copy element to first half
            ans[i] = nums[i];

            // Copy element to second half
            ans[i + len] = nums[i];
        }

        // Return the concatenated array
        return ans;
    }
}

File: Linear Shoal/Array_I/Q1. Concatenation of Array/Q1_ConcatenationofArray.py
class Solution(object):
    def getConcatenation(self, nums):
        # Create a new list to store the concatenated list
        answer = [0] * (2 * len(nums))
        j = 0

        # Iterate through the original list
        for i in range(len(nums)):
            # Add the current number to the new list
            answer[i] = nums[i]
            # Add the current number again at the corresponding position after the original numbers
            answer[i + len(nums)] = nums[i]

        # Return the concatenated list
        return answer
